Generate/general/xslt/generateTestScript.xsl

```diff
@@ -17,12 +17,15 @@
     <xsl:param name="commonComponentFolder" select="'../../general/common-tests'"/>
     
     <!-- The main template, which will call the remaining templates.
-         param inputDir is a required string where the XIS or PHR input dir resides. Inclusions will start relative to this base. -->
+         param inputDir is a string describing the direcory where the input file resides. The 
+                        'project/commonComponentFolder' template parameters are relative to this directory.
+                        This parameter is only needed when applying this template on a document that's not read from 
+                        disk (i.e. a generated/rewritten document). -->
     <xsl:template name="generate" match="f:TestScript">
-        <xsl:param name="inputDir" required="yes"/>
+        <xsl:param name="inputDir" as="xs:string" select="replace(base-uri(current()), '(.*)/[^/]+', '$1')"/>

```

Comment:
@jduwel , volgens mij doet dit hetzelfde als jouw tokenize-gebaseerde functie, toch?

Reply:
Ja, prima.
'substring-before-last' zou ik graag in een soort common functieset zou willen hebben als we ooit zover zijn, maar als je hem maar 1 keer gebruikt is het de facto niet nodig er een functie van te maken.

Reply:
Die common functieset heet FunctX toch?

Reply:
Ik had deze van Stackoverflow geïnterpreteerd, maar de overeenkomsten zijn flink :o.